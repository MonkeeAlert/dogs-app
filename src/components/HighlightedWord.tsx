import React from 'react';
import {Text, TextStyle, View} from 'react-native';

interface Props {
  text: string;
  highlight: string;
  style?: TextStyle | null;
  highlightStyle?: TextStyle | null;
}

export const HighlightedWord = ({
  text,
  highlight,
  style = null,
  highlightStyle = null,
}: Props) => {
  const parts = text
    .split(new RegExp(`(${highlight})`, 'gi'))
    .filter(i => i !== '');

  return (
    <View>
      <Text style={style}>
        {parts.map((i, k) =>
          i.toLowerCase() === highlight.toLowerCase() ? (
            <Text
              testID={`Highlighted`}
              key={`${text}-${k}`}
              style={highlightStyle ?? {fontWeight: '800'}}>
              {i}
            </Text>
          ) : (
            i
          ),
        )}
      </Text>
    </View>
  );
};
