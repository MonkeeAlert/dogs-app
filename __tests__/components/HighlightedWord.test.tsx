import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import HighlightedWord from '../../src/components/HighlightedWord';
import {TextStyle} from 'react-native';

const defaultProps = {
  text: '',
  highlight: '',
  style: null,
  highlightStyle: null,
};

const halfFilledProps = {
  text: 'foobarbaz',
  highlight: 'o',
};

const filledProps = {
  text: 'foobarbaz',
  highlight: 'o',
  style: {fontWeight: '500'} as TextStyle,
  highlightStyle: {backgroundColor: 'yellow'} as TextStyle,
};

describe('HighlightedWord', () => {
  it('matches snapshot', () => {
    const tree = renderer
      .create(<HighlightedWord {...defaultProps} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('highlights the word', () => {
    const wrapper = shallow(<HighlightedWord {...filledProps} />);

    const parts = wrapper.find({testID: 'Highlighted'});
    expect(parts.length).toBe(2);
  });

  it('style and highlightedStyle props are null', () => {
    const wrapper = shallow(<HighlightedWord {...halfFilledProps} />);

    const parts = wrapper.find({style: {fontWeight: '800'}});
    expect(parts.length).toBe(2);
  });
});
