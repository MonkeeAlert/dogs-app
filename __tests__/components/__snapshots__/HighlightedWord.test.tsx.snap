// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightedWord matches snapshot 1`] = `
<View>
  <Text
    style={null}
  />
</View>
`;
